app/ftpclient.py

```
@@ -0,0 +1,66 @@
+import datetime
+from ftplib import FTP
+from io import BytesIO
+import logging
+
+from publisher import Job
+
+
+class FTPWorker:
+
+    def __init__(self, user, password, host, port, timeout=30, **kwargs):
+        self.log = logging.getLogger("sdx.FTPWorker")
+        self.user, self.password = user, password
+        self.host, self.port = host, port
+        self.timeout = timeout
+        self.ftp = FTP()
+
+    def __enter__(self):
+        try:
+            self.ftp.connect(self.host, self.port, timeout=self.timeout)
+        except Exception as e:
```

Does this need to swallow all exceptions, or should we be more selective?

---

There may be different reasons for failure, but there isn't any choice of action we can take based on the type of exception. We either connect or we don't. If we don't, we can't proceed.